Close dropdown in intersection observer event method

dbauszus-glx opened the pull request:
This PR introduces the maxHeight property for dropdown elements.

This PR also introduces an intersectionObserver for the dropdown.node. The oberserver element will close the dropdown element if the position of the parent is changed [eg. from a scroll event].

simon-leech reviewed (dismissed):
If I open up a dropdown e.g. a filter near the bottom of the layer panel, previously the dropdown would open upwards so it wasn't lost off the page. 
On this PR, the arrow changes to indicate a click has been registered, but the dropdown cannot be seen. If i scroll so the dropdown is no longer at the bottom of the page, it can be seen. This should work as it did previously where a dropdown that's too large opens upwards. 

Dropdowns shown in a tabulator table are no longer visible either, the dropdown arrow changes but the content is not seen.

dbauszus-glx commented:
I fixed the issue in regards to the postioning and sizing of the dropdown.

https://github.com/user-attachments/assets/6aa12da3-4776-4b41-b31f-7b4b9ac169ba

I need to see the table dataview issue.

dbauszus-glx commented:
@simon-leech The positioning should be fixed for use in tabulator tables. The positioning must respect the mobile display where the offset may differ from the clientBoundingRect.top.

AlexanderGeere reviewed (changes requested):
I can still scroll the behind the dropdown in the multi-filter: 
![Screenshot from 2025-05-19 10-02-29](https://github.com/user-attachments/assets/2ebfa888-4738-4c89-a508-cc045a1f1bbd)

You can see this on `bugs_testing/filter/all_filters.json` the multi-filter is on the scratch layer.

dbauszus-glx commented:
Using a custom dropdown element is impossible difficult to position without an anchor css property which isn't yet fully formalized or supported.

The select / option element is positioned correctly but not fully stylable. The position is more important than the styling.

This PR introduces a select / option element for dropdowns.

We can adjust the styling in order to make this look like current with the limitation that the styling will not be fully supported by all platforms.

@AlexanderGeere The search will not work with a select/option element. Can you try to implement a native search element with the entries used to populate the <datalist> in a similar fashion like the select <options>. https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/search

AlexanderGeere commented:
@dbauszus-glx I have added a search element.

dbauszus-glx commented:
@AlexanderGeere This is looking pretty good already. I moved the searchInput method from it's own module into the dropdown module and unnested the onInput method as onSearchInput with the options assigned to the params object as searchOptions.

Is it possible to remove the dropdown icon which shows on hover over the search input?

Is it also possible to remove the duplicate item with the tick? Ideally a class should be set on the item which can be styled with a background colour.

![image](https://github.com/user-attachments/assets/f70d3d14-b801-4090-b7b4-82323e449739)

I removed the the dropdown copy now and also the dropdown style, and the multi_dropdown which is no longer used anywhere.

simon-leech reviewed (changes requested):
This isn't working properly for me. 
The styling looks different which I think will be something we should override. 

The main issue I'm seeing is that when I select an option from a dropdown list, the option is not highlighted in blue until I select the next option?

simon-leech commented:
![image](https://github.com/user-attachments/assets/f626caa3-14b5-47d1-9b67-3c64a3bb3ee8)

Dark mode styling needs correcting too

simon-leech commented:
![image](https://github.com/user-attachments/assets/c7fb1329-41e2-44d4-98f6-7fd1f607e62d)

Locale dropdown is empty on load too

dbauszus-glx commented:
I updated the background-color and color for the select element to be applied for dark_mode.

dbauszus-glx commented:
The locale dropdown uses the lagacy span property for the placeholder. I have updated this.

simon-leech commented:
@AlexanderGeere  If I type into a dropdown_pills entry I can end up with an empty pill being built. 
![image](https://github.com/user-attachments/assets/7b2ef55a-4976-4b89-85c7-de50e18bbc4e)

To do this I just type letters and then delete them, and it adds an empty pill which shouldn't happen.

simon-leech reviewed (approved):
Good with me - think I've tested all the dropdown locations 🤞😅

AlexanderGeere reviewed (approved):
This looks good to me.
